testsuite/fuzzer/fuzz/fuzz_targets/move/aptosvm_authenticators.rs

```diff
@@ -0,0 +1,363 @@
+
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_cached_packages::aptos_stdlib;
+use aptos_crypto::{
+    ed25519::{Ed25519PrivateKey, Ed25519PublicKey},
+    PrivateKey, SigningKey, Uniform,
+};
+use aptos_language_e2e_tests::{
+    account::Account, data_store::GENESIS_CHANGE_SET_HEAD, executor::FakeExecutor,
+};
+use aptos_types::{
+    chain_id::ChainId, keyless::{
+        AnyKeylessPublicKey, EphemeralCertificate, IdCommitment, KeylessPublicKey, KeylessSignature, OpenIdSig, Pepper, TransactionAndProof
+    }, move_any::Any, transaction::{
+        authenticator::{
+            AccountAuthenticator, AnyPublicKey, AnySignature, EphemeralPublicKey,
+            EphemeralSignature, SingleKeyAuthenticator, TransactionAuthenticator,
+        },
+        ExecutionStatus, SignedTransaction, TransactionStatus,
+    }, write_set::WriteSet
+};
+use aptos_vm::AptosVM;
+use libfuzzer_sys::{fuzz_target, Corpus};
+use move_core_types::vm_status::{StatusCode, StatusType};
+use once_cell::sync::Lazy;
+use std::sync::Arc;
+mod utils;
+use utils::{check_for_invariant_violation, FuzzerTransactionAuthenticator, Style, TransactionState};
+
+// genesis write set generated once for each fuzzing session
+static VM: Lazy<WriteSet> = Lazy::new(|| GENESIS_CHANGE_SET_HEAD.write_set().clone());
+
+const FUZZER_CONCURRENCY_LEVEL: usize = 1;
+static TP: Lazy<Arc<rayon::ThreadPool>> = Lazy::new(|| {
+    Arc::new(
+        rayon::ThreadPoolBuilder::new()
+            .num_threads(FUZZER_CONCURRENCY_LEVEL)
+            .build()
+            .unwrap(),
+    )
+});
+
+fn run_case(input: TransactionState) -> Result<(), Corpus> {
+    tdbg!(&input);
+
+    AptosVM::set_concurrency_level_once(FUZZER_CONCURRENCY_LEVEL);
+    let mut vm = FakeExecutor::from_genesis_with_existing_thread_pool(
+        &VM,
+        ChainId::mainnet(),
+        Arc::clone(&TP),
+    )
+    .set_not_parallel();
+
+    let sender_acc = if true {
+        // create sender pub/priv key. initialize and fund account
+        vm.create_accounts(1, input.tx_auth_type.sender().fund_amount(), 0)
+            .remove(0)
+    } else {
+        // only create sender pub/priv key. do not initialize
+        Account::new()
+    };
+
+    let receiver = Account::new();
+
+    // build tx
+    let tx = sender_acc
+        .transaction()
+        .payload(aptos_stdlib::aptos_coin_transfer(*receiver.address(), 1))
+        .sequence_number(0)
+        .gas_unit_price(100)
+        .max_gas_amount(1000);
+
+    let tx_auth_type = input.tx_auth_type.clone();
+
+    let raw_tx = tx.raw();
+    let tx = match tx_auth_type {
+        FuzzerTransactionAuthenticator::Ed25519 { sender: _ } => raw_tx
+            .sign(&sender_acc.privkey, sender_acc.pubkey.as_ed25519().unwrap())
+            .map_err(|_| Corpus::Keep)?
+            .into_inner(),
+        FuzzerTransactionAuthenticator::Keyless {
+            sender: _,
+            style,
+            any_keyless_public_key,
```

Comment:
How is this sampled? Via the `Arbitrary` trait?

How do you plan on sampling keyless public keys and signatures that are at least partially-valid (e.g., the `EphemeralCertificate` in the `KeylessSignature` verifies under the `KeylessPublicKey`?)

Reply:
I guess you will modify them appropriately inside your `match style` cases, as needed for that fuzzing strategy?

Reply:
Yes, I can override within the match what I want done in a specific way.